src/main/scala/org/apache/mesos/chronos/scheduler/api/DependentJobResource.scala

```diff
@@ -57,10 +57,16 @@ class DependentJobResource @Inject()(
         val parents = jobGraph.parentJobs(newJob)
         oldJob match {
           case j: DependencyBasedJob =>
-            val oldParents = jobGraph.parentJobs(j)
-            oldParents.map(x => jobGraph.removeDependency(x.name, oldJob.name))
+            val newParentNames = parents.map(_.name)
+            val oldParentNames = jobGraph.parentJobs(j).map(_.name)
+
+            if (newParentNames != oldParentNames) {
+              oldParentNames.foreach(jobGraph.removeDependency(_, oldJob.name))
+              newParentNames.foreach(jobGraph.addDependency(_, newJob.name))
+            }
             jobScheduler.removeSchedule(j)
           case j: ScheduleBasedJob =>
```

Comment:
If this is done then the bug is reintroduced because `JobGraph.addDependency` [resets](https://github.com/mesos/chronos/blob/master/src/main/scala/org/apache/mesos/chronos/scheduler/graph/JobGraph.scala#L97) the counter of `edgeInvocationCount`.
Adding dependencies is executed for both type of jobs.
`newParentNames.foreach(jobGraph.addDependency(_, newJob.name))` does it for a `DependencyBasedJob` and

```
case j: ScheduleBasedJob =>
  parents.foreach(p => jobGraph.addDependency(p.name, newJob.name))
```
does it for a former-`ScheduleBasedJob`-now-`DependencyBasedJob`.

It's just that, for a `DependencyBasedJob` that is updated and is not converted from a `ScheduleBasedJob`, the update of its dependencies only happens if the dependencies actually changed.